src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsBuilderExtensions.cs

```diff
@@ -58,22 +62,77 @@ public static IWebJobsBuilder UseHostId(this IWebJobsBuilder builder, string hos
         }
 
         public static IWebJobsBuilder UseWebJobsStartup(this IWebJobsBuilder builder, Type startupType)
+        {
+            return builder.UseWebJobsStartup(startupType, NullLoggerFactory.Instance);
+        }
+
+        public static IWebJobsBuilder UseWebJobsStartup(this IWebJobsBuilder builder, Type startupType, ILoggerFactory loggerFactory)
         {
             if (!typeof(IWebJobsStartup).IsAssignableFrom(startupType))
             {
                 throw new ArgumentException($"The {nameof(startupType)} argument must be an implementation of {typeof(IWebJobsStartup).FullName}");
             }
 
             IWebJobsStartup startup = (IWebJobsStartup)Activator.CreateInstance(startupType);
-            startup.Configure(builder);
 
+            if (loggerFactory == NullLoggerFactory.Instance)
+            {
+                startup.Configure(builder);
+            }
+            else
+            {
+                ConfigureAndLogUserConfiguredServices(startup, builder, loggerFactory);
+            }
             return builder;
         }
 
+        private static void ConfigureAndLogUserConfiguredServices(IWebJobsStartup startup, IWebJobsBuilder builder, ILoggerFactory loggerFactory)
+        {
+            var logger = loggerFactory.CreateLogger<ServiceCollectionDecorator>();
+
+            if (builder.Services is ITrackedServiceCollection tracker)
+            {
+                if (tracker != null)
+                {
+                    startup.Configure(builder);
+                    StringBuilder sb = new StringBuilder("Services registered by external startup type " + startup.GetType().ToString() + ":");
+
+                    foreach (ServiceDescriptor service in tracker.TrackedCollectionChanges)
+                    {
+                        string serviceName = "Service name: " + service.ServiceType.FullName;
+                        StringBuilder registeredService = new StringBuilder(", Implementation: ");
+                        if (service.ImplementationType != null)
+                        {
+                            registeredService.Append(service.ImplementationType.FullName);
+                        }
+                        else if (service.ImplementationInstance != null)
+                        {
+                            registeredService.Append(service.ImplementationInstance.GetType().FullName + " [instance]");
+                        }
+                        else if (service.ImplementationFactory != null)
+                        {
+                            registeredService.Append("Using ImplementationFactory");
+                        }
+
+                        string lifetime = ", Lifetime: " + service.Lifetime.ToString();
+                        sb.Append("\n\t" + serviceName + registeredService.ToString() + lifetime);
+                    }
+                    logger.LogDebug(sb.ToString(), new EventId(500, "ExternalStartupServices"));
```

Comment:
`EventId` needs to be the first parameter here. -- there's an overload that allows for `object` to be passed, which is why this is working, but the event wouldn't get picked up by kusto.